lib/httpServer/transports/http/index.js

```diff
@@ -615,6 +637,48 @@ function safeDecodePath(path) {
 	}
 }
 
+/**
+ * Encapsulate the request object
+ *
+ * This function currently only takes care of
+ * checking the HTTP POST size.
+ *
+ * @param {*} req HTTP request
+ */
+function setRequestFilters(req) {
+	var realOn = req.on.bind(req);
+	var byteSize = 0;
+
+	function isRequestTooLarge() {
+		return byteSize > maxRequestSize;
+	}
+
+	req.on('data', function (chunk) {
+		byteSize += chunk.length;
+
+		if (isRequestTooLarge()) {
+			req.emit('error', new RequestError('Entity too large', req));
+			req.removeAllListeners();
+		}
+	});
+
+	req.on = function (eventName, func) {
```

Comment:
Why this hackery? Can't we just listen along on `req` and destroy the ongoing request when we reach the threshold?